al-nginx/nginx.conf

```diff
@@ -42,92 +42,38 @@ http {
             root /usr/share/nginx/html;
         }
 
-        location = /modelujeme/sluzby/auth-server {
-          return 302 /modelujeme/sluzby/auth-server/;
-        }
-
         location /modelujeme/sluzby/auth-server/ {
-            proxy_pass http://al-auth-server:8080;
-            proxy_set_header X-Forwarded-Host $host;
+            proxy_pass http://al-auth-server:8080; # no trailing slash to preserve matched prefix
+            # Keycloak does not work without the settings below for some reason
             proxy_buffer_size   128k;
             proxy_buffers       4 256k;
             proxy_busy_buffers_size 256k;
         }
 
-        location = /modelujeme/sluzby/db-server {
-            return 302 /modelujeme/sluzby/db-server/;
-        }
-
         location /modelujeme/sluzby/db-server/ {
-            rewrite ^/modelujeme/sluzby/db-server/(.*) /$1 break;
-            proxy_pass http://al-db-server:7200;
-            proxy_redirect http://al-db-server:7200/ /modelujeme/sluzby/db-server/;
-            proxy_set_header X-Forwarded-Host $host;
-        }
-
-        location = /modelujeme/sluzby/sgov-server {
-          return 302 /modelujeme/sluzby/sgov-server/;
+            proxy_pass http://al-db-server:7200/; # keep the trailing slash to cut off matched prefix
         }
 
         location /modelujeme/sluzby/sgov-server/ {
-            proxy_pass http://al-sgov-server:8080/modelujeme/sluzby/sgov-server/;
-            proxy_set_header X-Forwarded-Host $host;
-        }
-
-        location = /modelujeme/sluzby/termit-server {
-          return 302 /modelujeme/sluzby/termit-server/;
+            proxy_pass http://al-sgov-server:8080; # # no trailing slash to preserve matched prefix
```

Comment:
Termit-server and sgov-server are configured to accept the whole path (e.g. `modelujeme/sluzby/termit-server`) similar to Keycloack, but it would be very easy to get rid of the prefix as it is unnecessary.